pkg/metric/middleware_test.go

```diff
@@ -0,0 +1,86 @@
+package metric
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"strconv"
+	"testing"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"gotest.tools/assert"
+)
+
+func TestDurationCode(t *testing.T) {
+
+	registry := prometheus.NewRegistry()
+
+	// mock http handler:
+	httpHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {})
+
+	//mock HistogramVec
+	durationHistogram := promauto.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name: "http_request_duration_seconds_test",
+			Help: "Duration of my request in seconds.",
+		},
+		[]string{"code", "query_type"},
+	)
+	registry.MustRegister(durationHistogram)
+
+	//add mock label
+	duration, _ := durationHistogram.CurryWith(prometheus.Labels{"query_type": "mock_query"})
+
+	// mock instrumentHandlerDuration with mock observervec
+	instrumentedHandler := promhttp.InstrumentHandlerDuration(
+		duration,
+		httpHandler,
+	)
```

Comment:
This instrumentation is done inside `PrometheusMiddleware()` so the test shouldn't add it. Instead we need to confirm that `PrometheusMiddleware()` is adding it.

Reply:
Oh right! Thanks  - will update test